anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -2177,6 +2199,37 @@ def add_or_update_image(
     return ret
 
 
+def is_new_tag(image_record: dict, registry: str, repo: str, tag: str):
+    """
+    Returns false if the provided registry, repo, tag tuple exactly match those fields in one of the image_detail objects in
+    image_record. If no such match is possible, return true.
+    """
+
+    # If we are missing an image_details block entirely or it is None, assume this is a new tag
+    if "image_detail" not in image_record or not image_record["image_detail"]:
+        return True
+    else:
+        # Iterate through each existing image_detail
+        for image_detail in image_record["image_detail"]:
+            # We shouldn't be missing one of these fields, but if we are, assume this is not a match
+            if (
+                "registry" not in image_detail
+                or "repo" not in image_detail
+                or "tag" not in image_detail
+            ):
+                return True
```

Comment:
I think it should continue iterating through the list here rather than returning True if registry repo or tag are not in the image_detail. To make this simpler I would recommend the loop look like this 
```
for image_detail in image_record.get("image_detail", []):
  existing_registry = image_detail.get("registry")
  existing_repository = image_detail.get("repo")
  existing_tag = image_detail.get("tag")

   if existing_registry == registry and existing_repository == repository and existing_tag == tag:
     return False  
else:
  return  True
```

Reply:
I made that change as well. This is better

Reply:
As now if one of those parameters is missing in both tags, it will match correctly (albeit that would be very unexpected)